lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -368,6 +365,18 @@ async sub transform_transaction {
             timestamp    => $int_timestamp,
         );
 
+        # if the gas is empty we don't proceed
+        unless ($gas && $decoded_transaction->{gasPrice}) {
```

Comment:
is there any possible changes on `$decoded_transaction`, I mean since we have `$gas` equal to that in #L348

Reply:
not sure if i got you correctly.... the reason of moving the check here is to be able to use `$transaction` in `_set_trasaction_type` in #L370.